File: src/commons/enums.py
from enum import Enum


class GAME_STATES(Enum):
    # in this state, all bots should go to home position
    # and the game clock is not running
    game_over = 0
    # in this state, bots that are in autonomous mode will
    # play the game and the clock is running
    game_on = 1
    # in this state, all bots should stop where they are and
    # the clock should be paused
    game_paused = 2
    # in this state, all bots should return to home
    going_home = 3


VALID_GAME_STATES = set(item.value for item in GAME_STATES)


class BOT_INDEXES(Enum):
    ball_bot = 0
    player_1 = 1
    player_2 = 2


VALID_BOT_INDEXES = set(item.value for item in BOT_INDEXES)


class BOT_MODES(Enum):
    auto = 0
    manual = 1


VALID_BOT_MODES = set(item.value for item in BOT_MODES)

File: src/scoreboard/src/selectors/enums.js
export const GAME_STATES = {
  game_over: 0,
  game_on: 1,
  game_paused: 2,
  return_home: 3,
};

export const GAME_STATE_LABELS = [
  "Game Over",
  "Game On!",
  "Game Paused",
  "Returning to Home",
];
export const VALID_GAME_STATES = Object.values(GAME_STATES);

export const BOT_INDEXES = {
  ball_bot: 0,
  player_1: 1,
  player_2: 2,
};
export const VALID_BOT_INDEXES = Object.values(BOT_INDEXES);

export const BOT_MODES = {
  auto: 0,
  manual: 1,
};

export const BOT_MODE_NAMES = ["Autonomous", "Manual"];

export const VALID_BOT_MODES = Object.values(BOT_MODES);
